safe/impact_functions/impact_function_metadata.py

```diff
@@ -147,8 +147,9 @@ def allowed_subcategories(cls, category=None):
                 .allowed_subcategories('hazard')
         else:
             metadata_dict = cls.as_dict()
-            categories = metadata_dict['categories']
-            result = add_to_list(result, categories[category]['subcategories'])
+            categories = metadata_dict['layer_requirements']
+            result = add_to_list(result,
+                                 categories[category]['%s_types' % category])
             return result
```

Comment:
I think we don't use this function anymore right? If so, I will delete the function. Some apply for another functions in impact function metadata and manager.